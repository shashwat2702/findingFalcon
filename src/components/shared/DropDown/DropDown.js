/* eslint-disable react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import './DropDown.scss';

const DropDown = ({
  planets, onChange, name, selectedPlanets,
}) => (
  <div className="dropDown">
    <select value={selectedPlanets[Number(name) - 1]} onChange={onChange} name={name}>
      <option value="Select One">Select One</option>
      {(selectedPlanets[Number(name) - 1] !== '')
      && (
      <option
        value={selectedPlanets[Number(name) - 1]}
        key={selectedPlanets[Number(name) - 1]}
      >
        {selectedPlanets[Number(name) - 1]}

      </option>
      )
      }
      {
          planets.map(planet => (!selectedPlanets.includes(planet.name)
            && <option value={planet.name} key={planet.name}>{planet.name}</option>
          ))
      }
    </select>
  </div>
);
DropDown.propTypes = {
  planets: PropTypes.array.isRequired,
  onChange: PropTypes.func.isRequired,
  name: PropTypes.number.isRequired,
  selectedPlanets: PropTypes.array.isRequired,
};
export default DropDown;
