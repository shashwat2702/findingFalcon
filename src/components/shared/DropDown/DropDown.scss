select {
    width: 100%;
    height: 2rem;
    font-size: 1.5rem;
    min-width: 150px;
    background-color: white;
}