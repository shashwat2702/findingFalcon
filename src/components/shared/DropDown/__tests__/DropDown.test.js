import React from 'react';
import renderer from 'react-test-renderer';
import DropDown from '../DropDown';


jest.mock('react-router-dom');

const planets = [{ name: 'PlanetA', distance: 100 }, { name: 'PlanetB', distance: 150 },
  { name: 'PlanetC', distance: 200 }, { name: 'PlanetD', distance: 50 },
  { name: 'PlanetE', distance: 250 }, { name: 'PlanetF', distance: 300 }];
const onChangeMocked = jest.fn();
const name = 1;
const selectedPlanets = ['PlanetC', 'PlanetF', 'PlanetD', 'PlanetA'];


describe('DropDown snapshot', () => {
  it('should render correctly', () => {
    const wrapper = renderer.create(<DropDown
      planets={planets}
      onChange={onChangeMocked}
      name={name}
      selectedPlanets={selectedPlanets}
    />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
