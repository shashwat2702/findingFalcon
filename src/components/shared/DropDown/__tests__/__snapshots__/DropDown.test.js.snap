// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown snapshot should render correctly 1`] = `
<div
  className="dropDown"
>
  <select
    name={1}
    onChange={[MockFunction]}
    value="PlanetC"
  >
    <option
      value=""
    >
      Select One
    </option>
    <option
      value="PlanetC"
    >
      PlanetC
    </option>
    <option
      value="PlanetB"
    >
      PlanetB
    </option>
    <option
      value="PlanetE"
    >
      PlanetE
    </option>
  </select>
</div>
`;
